### DISCUSSION TITLE
Implications on slowness ? -> JSON embed buffers are slow! consider using glTF binary

### DISCUSSION THREAD

**Discussion by ThanosRestas** [atteneder/glTFast#383] [Q&A] [answered]
What does _slow_ mean in this context ? Does anyone have any metrics ?

**Comment by atteneder** [comment, marked as answer]
No exact metrics, but here's the additional load from the top of my head:

1. 33% larger buffers (from [base64](https://en.wikipedia.org/wiki/Base64) overhead)
2. The JSON parser will make a copy when creating a managed string from the JSON field
3. Another copy of the managed string is made when separating the actual data from the meta prefix (string.SubString)
4. For the final buffer, additional memory is allocated.
5. Conversion is done on main thread.

Some of this could be optimized, but I heavily focused on optimizing for regular/binary files so far.

I don't have exact metrics at hand, though it would be interesting! The larger the meshes/vertices-/triangle counts (and thus glTF's buffers) the larger difference. For smallish objects there won't be much of a difference.

To make exact metrics, you could make embed and glTF-binary variants of the same file, load them in a [performance test](https://docs.unity3d.com/Packages/com.unity.test-framework.performance@2.8/manual/index.html) and compare the numbers.

hth

**Reply by ThanosRestas** [reply]
Thanks for the concise answer. When I test and record the metrics I will report back, it will be fun !
